provisioning/baremetal_pod.go

```diff
@@ -798,13 +782,29 @@ func EnsureMetal3Deployment(info *ProvisioningInfo) (updated bool, err error) {
 		info.EventRecorder, metal3Deployment, expectedGeneration)
 	if err != nil {
 		err = fmt.Errorf("unable to apply Metal3 deployment: %w", err)
-		return
+		// Check if ApplyDeployment failed because the existing Pod had an outdated
+		// Pod Selector.
+		selector, get_err := getMetal3DeploymentSelector(info.Client.AppsV1(), info.Namespace)
+		if get_err != nil || selector == metal3Deployment.Spec.Selector {
+			return
+		}
+		// This is an older deployment with the incorrect Pod Selector.
+		// Delete deployment and re-create
+		_ = DeleteMetal3Deployment(info)
```

Comment:
Any particular reason to ignore the error here?

Reply:
If the delete didn't succeed, the error from the create would fail with an error that the resource already exists. So, in the specific case, I felt it was OK to not specifically report the error with the delete because it was immediately followed by a create.

Reply:
We won't see the reason that the delete didn't succeed in the logs if we don't return it.

Reply:
Well, the create's error would say that the create failed because the metal3 deployment already exists. That way we would know the delete succeeded.